### ISSUE TITLE
Dynamic Routes

### ISSUE THREAD

**Issue by luigiminardim** [ayonli/next-controller#2]
I don't see in the documentation how can I get a route param. Is there a way to get this dynamic params? Something like: 

```ts
// pages/api/[name]/hello.ts

@api
export default class Controller extends ApiController {
 
 override async get(query) {
    const name = query.name;
    return `hello ${name}`
  }

}
```

**Comment by hyurl** [comment]
You can access the route params from this.req.params, HTTP related methods only accept query string params.

**Comment by luigiminardim** [comment]
@hyurl, thanks a lot for your answer. I have used your work on some of my projects and I liked it a lot.

 But, about your answer, I'm receiving a type error when trying to use `this.req.params`.
```
Property 'params' does not exist on type 'IncomingMessage'
```

**Comment by hyurl** [comment]
If you're using the default Next.js server, it seems it binds route params to the `query` object directly, which will be overridden by the previous **next-controller**, so I've just published a fix (version **1.5.2**) that merges the traditional query object and the route params, upgrade to this version and you can now access those params from the `query` object as Next.js' default behavior.

**Comment by hyurl** [comment]
@luigiminardim If you're using the default Next.js server, it seems it binds route params to the `query` object instead, which will be overridden by the previous next-controller, so I've just published a fix (version **1.5.2**) that merges the traditional query object and the route params, upgrade to this version and you can now access those params from the query object as Next.js' default behavior.

**Comment by luigiminardim** [comment]
@hyurl, Worked really well. Thanks for your help!
